components/editor/modules/dynamiccomponent/EditOverlay.js

```diff
@@ -21,25 +21,93 @@ export default (props) => (
     }}>
     {({data, onChange}) => {
       const config = data.toJS()
+      const parent = props.editor.value.document.getParent(props.node.key)
```

Comment:
I wouldn't know how do it differently right now, but we in general have a lot of parent sniffing going on. It just doesn't help reusability a lot.